File: clonewhatapp/CloneWhatsApp.API/Models/Llamada.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CloneWhatsApp.API.Models;

public enum TipoLlamada
{
    Voz,
    Video
}

public enum EstadoLlamada
{
    EnCurso,
    Completada,
    Perdida,
    Rechazada
}

public class Llamada
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public Guid LlamadorId { get; set; }

    [Required]
    public Guid ReceptorId { get; set; }

    public TipoLlamada Tipo { get; set; }

    public EstadoLlamada Estado { get; set; } = EstadoLlamada.EnCurso;

    public DateTime FechaInicio { get; set; } = DateTime.UtcNow;

    public DateTime? FechaFin { get; set; }

    public int? DuracionSegundos { get; set; }

    // Navigation properties
    [ForeignKey("LlamadorId")]
    public Usuario Llamador { get; set; } = null!;

    [ForeignKey("ReceptorId")]
    public Usuario Receptor { get; set; } = null!;
}

File: clonewhatapp/whatsapp-clone-android/domain/src/main/java/com/clonewhatsapp/domain/model/Llamada.kt
package com.clonewhatsapp.domain.model

data class Llamada(
    val id: String,
    val llamanteId: String,
    val receptorId: String,
    val tipo: TipoLlamada,
    val estado: EstadoLlamada,
    val fechaInicio: Long,
    val duracionSegundos: Int? = null,
    val nombreContacto: String? = null,
    val fotoContacto: String? = null,
)

enum class TipoLlamada {
    VOZ,
    VIDEO,
}

enum class EstadoLlamada {
    EN_CURSO,
    COMPLETADA,
    PERDIDA,
    RECHAZADA,
}
